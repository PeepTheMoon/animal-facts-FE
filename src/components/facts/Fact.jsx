import React from 'react';
import PropTypes from 'prop-types';

const Fact = ({ image, caption }) => (
  <figure>
    <img src={image} />
    <figcaption>{caption}</figcaption>
  </figure>
);

Fact.propTypes = {
  image: PropTypes.string.isRequired,
  caption: PropTypes.string.isRequired
};

export default Fact;

