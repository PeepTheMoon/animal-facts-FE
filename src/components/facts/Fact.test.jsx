import React from 'react';
import { shallow } from 'enzyme';
import Fact from './Fact';

describe('Fact component', () => {
  it('renders Fact', () => {
    const wrapper = shallow(<Fact image='image.png' caption='wild animals' />);
    expect(wrapper).toMatchSnapshot();
  });
});
