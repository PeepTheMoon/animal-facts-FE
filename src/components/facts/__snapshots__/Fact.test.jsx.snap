// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fact component renders Fact 1`] = `
<figure>
  <img
    src="image.png"
  />
  <figcaption>
    wild animals
  </figcaption>
</figure>
`;
